KeyVault.Acmebot/Startup.cs

```diff
@@ -85,6 +86,7 @@ public override void Configure(IFunctionsHostBuilder builder)
             dnsProviders.TryAdd(options.Gandi, o => new GandiProvider(o));
             dnsProviders.TryAdd(options.GoDaddy, o => new GoDaddyProvider(o));
             dnsProviders.TryAdd(options.GoogleDns, o => new GoogleDnsProvider(o));
+            dnsProviders.TryAdd(options.Namecheap, o => new NamecheapProvider(o, HttpClientFactory.Create(), HttpClientFactory.Create()));
```

Comment:
HttpClient should be instantiated inside the Provider. Implementation uniformity with other Providers is important.